### PR TITLE
Move .col-* classes to same media query after .row-cols-* classes

### LINKED ISSUE #34335: v5.0.2 regression: grid col-N now overrides col-breakpoint
Between v5.0.1 and v5.0.2, the col-breakpoint declarations were moved above the col-N declarations. This breaks automatic column sizing. For example, a class declaration of "col-6 col-md" now creates a column that is always six columns wide and the column width does not become automatic at the breakpoint. Previously, the column width was six columns at the small sizes, and became automatic at the breakpoint.

Here is the diff showing the change in the compiled CSS: https://github.com/twbs/bootstrap/commit/688bce4fa695cc360a0d084e34f029b0c192b223#diff-1e4f2454c1c043f0cf31f56b51ec891c6bc0ff1c88985b82c7f89c0d8f78c4ba

This **might** be a result of this commit, although my guess is based mainly on the title of the commit, which describes changing the declaration order for cols: https://github.com/twbs/bootstrap/commit/f2b47e1c8a263f4406255991af9fd55984c22a0e#diff-41667d8b9901aa9fa52483b538bb9026c287f2c663d2fdc01acffa06888cc087

### PR DESCRIPTION
**Pull request by mdo**
Fixes #34335.

### CODE REVIEW

**Review by angrymouse — APPROVED** [review]

**Review by danielkauffman — COMMENTED** [review]
Fixes #34335 awesome!

CodePen showing the issue in v5.0.2: https://codepen.io/rocksolidsolutions/pen/zYZVeeq

CodePen showing the fix in this branch: https://codepen.io/rocksolidsolutions/pen/qBrzgKq

Building the dist files and running `git diff v5.0.1 -- dist/css/bootstrap-grid.css` and `git diff v5.0.2 -- dist/css/bootstrap-grid.css` both seem okay.

I suppose it would be good to check that #33621 is still fixed after this change.

**Review by danielkauffman — COMMENTED** [review]
Per @lacutah #33530 expects `row-cols*` to override `col`. I theoretically like the cleanliness of having all of the `row-cols*` declarations together and all of the `col` declarations together (as with this PR) but doing so either breaks `cols row-cols` or breaks `col-N col-breakpoint`. I'm not sure how widespread each of these use cases are. But the simple solution (assuming both cases need to be supported) seems to be ordering the declarations first `col` then `row-cols*` then `col-breakpoint*`.

**Comment by ffoodd** [comment]
I agree with @danielkauffman here. We have an issue with this PR regarding `.col` (without breakpoint): see #33621 [CodePen using this PR's CSS](https://codepen.io/ffoodd/pen/VwbwRdW) to reproduce.

`.col` should always be first, and `.col-*` last.

File: scss/mixins/_grid.scss [hunk 1]
```diff
@@ -92,6 +87,11 @@
     $infix: breakpoint-infix($breakpoint, $breakpoints);
 
     @include media-breakpoint-up($breakpoint, $breakpoints) {
+      // Provide basic `.col-{bp}` classes for equal-width flexbox columns
+      .col#{$infix} {
```
Comment (ffoodd) [inline scss/mixins/_grid.scss:91]: IMHO we should skip if `$infix` is empty, and prefer to set `.col` before any loop. Maybe hard-coded in our main grid file, maybe?

**Review by ffoodd — CHANGES_REQUESTED** [review]

**Comment by mdo** [comment]
Punting on this for #34612 right now.

**Pull request closed without merging**